### ISSUE TITLE
invalid data type

### ISSUE THREAD

**Issue by npravecek** [catboost/catboost#1514]
Problem: cant figure out how to load training set 
catboost version: 0.24.3
Operating System: ubuntu 20.04
CPU: i7 4790k
# GPU: gtx 1070

Here is my code 

y = train_df['answered_correctly']
columns = ['mean_user_accuracy', 'questions_answered', 'mean_accuracy', 'question_asked',
           'prior_question_had_explanation',# 'mean_diff1', 'mean_diff2', 'mean_user_content_accuracy'
          ]
X = train_df[columns]

folds = StratifiedKFold(n_splits=5, shuffle=False)
X_aa = X.as_gpu_matrix()
y_aa = cupy.array(y).get()
trainyboi = cudf.DataFrame(train_df[features])
trainyboilabel = cudf.DataFrame(train_df[target])
for fold_n, (train_index, valid_index) in enumerate(folds.split(X_aa, y_aa)):
    print(f'Fold {fold_n} started at {time.ctime()}')
    train_set = Pool(trainyboi, trainyboilabel)
    val_set = Pool(valid_df[features], label = valid_df[target])
    model = CatBoostClassifier(**params)
    model.fit(train_set, val_set)
    score = max(model.evals_result_['valid_1']['auc'])
    
    models.append(model)
    scores.append(score)

    fold_importance = cudf.DataFrame()
    fold_importance["feature"] = columns
    fold_importance["importance"] = model.feature_importances_
    fold_importance["fold"] = fold_n + 1
    feature_importance = cudf.concat([feature_importance, fold_importance], axis=0)
    break
I can show more code if this is not the issue. Thanks in advance for any help. I did look extensively online for help before this but I couldn't figure out this exact issue. Also I am using rapids cudf to speed up the data process.

**Comment by kizill** [comment]
Hello!
Currently CatBoost don't support cudf dataframes, so if it's possible to give CatBoost regular CPU Pandas DataFrame - everything should work just fine!
We need some information for understanding - do we really need to invest into CUDF support. Could you share some information about why are you using CUDF and how important CUDF support in CatBoost could be for you?

**Comment by npravecek** [comment]
I was using it to increase performance for a kaggle project and cudf is
pretty important because xgboost has it and I ended switching my project
over to it for that feature

**Comment by btbpanda** [comment]
Hello!
I'm developing opensource LightAutoML solution in Sber AI Lab (https://github.com/sberbank-ai-lab/LightAutoML). We are using CatBoost as a part of the pipeline. Right now we are building end-to-end GPU pipeline and it would be very helpful if you support direct loading from cupy ndarray and cudf. It will help to avoid extra copies and memory leakage and speed up CatBoost preprocessing part (if it's possible to be made on GPU). CatBoost does long CPU preprocessing part before to start GPU training and some times if dataset is big enough, this could actually be the longest stage.
Thanks

**Comment by RishatZagidullin** [comment]
Hello!

Here to support the idea of adding the option of passing GPU data directly to Catboost. We are preparing study material for HPC courses at Skoltech, and one of the topics concerns Gradient Boosting frameworks. In terms of the speed Catboost is one of the best solutions but preprocessing on GPU is hindered by the CPU-GPU transfers. 

I was comparing the performance of Catboost to the distributed XGBoost implementation from [here](https://developer.nvidia.com/blog/accelerating-xgboost-on-gpu-clusters-with-dask/) on the dataset of 1.2 GB and the speeds are very comparable. But with bigger data and a more complex pipeline, the Catboost based pipeline would suffer from host-device data transfers which can be avoided.

Also, it's related to issue #1404. With the Dask Dataframe support Catboost would enable users to preprocess data on the GPUs in a distributed fashion.
